packages/http-caching-proxy/src/__tests__/integration/http-caching-proxy.integration.ts

```diff
@@ -4,20 +4,41 @@
 // License text available at https://opensource.org/licenses/MIT
 
 import {expect} from '@loopback/testlab';
+import axios, {AxiosProxyConfig, AxiosRequestConfig} from 'axios';
 import delay from 'delay';
 import http from 'http';
 import {AddressInfo} from 'net';
 import pEvent from 'p-event';
 import path from 'path';
-import makeRequest from 'request-promise-native';
 import rimrafCb from 'rimraf';
+import tunnel, {ProxyOptions as TunnelProxyOptions} from 'tunnel';
+import {URL} from 'url';
 import util from 'util';
 import {HttpCachingProxy, ProxyOptions} from '../../http-caching-proxy';
 
 const CACHE_DIR = path.join(__dirname, '.cache');
 
 const rimraf = util.promisify(rimrafCb);
 
+const axiosInstance = axios.create({
+  // Provide a custom function to control when Axios throws errors based on
+  // http status code. Please note that Axios creates a new error in such
+  // condition and the original low-level error is lost
+  validateStatus: () => true,
```

Comment:
I am not very happy about this part, it looks rather hacky to me. I also dislike the fact that when using Axios, we must be remember to modify the default configuration if we want to get decent error handling. I am pretty sure we won't be consistent and in some places we will forget to do the necessary tweaks, thus making it difficult for our users to troubleshoot request errors they may encounter in production. What's worse, when they fill an issue about the problem, it's unlikely that we will remember by that time what's the culprint and how to fix Axios configuration.

Reply:
I am taking back the comment above. I checked Axios' source code and run our tests with DEBUG statements enabled, and both checks confirmed that Axios is preserving original error details & stack traces when an error is reported by Node.js core.

Here is the relevant code: https://github.com/axios/axios/blob/03e6f4bf4c1eced613cf60d59ef50b0e18b31907/lib/adapters/http.js#L224-L245

As I understand `validateStatus`, it determines whether a successfully received HTTP response is converted into a resolved or a rejected promise, depending on the status code. (I.e. whether `await axios(/*...*/)` should throw an error when the server responds with `500 Internal Server Error`.)